my-app/src/utils/leafletDrawUtils.ts

```diff
@@ -0,0 +1,52 @@
+// src/utils/leafletDrawUtils.ts
```

Comment:
The whole code surrounding leaflet draw seems like way too much. We talked about geoman and how you should use it to manipulate layers on the map. Overhaul this implementation to use geoman, no need to build the whole editing logic from the bottom up and define it explicitly. What if you had more than just the 1 type of polygon (Must entrance for example or an area where certain rules apply instead of just no entrance)